src/legacyilibassemble.js

```diff
@@ -183,64 +186,123 @@ function assemblejs() {
                         readData = readFile(jsonPath);
-                        if (readData) outFile[lang]["ilib.data." + jsonName + "_und_" + region] = readData;
+                        if (readData) {
+                            key = "ilib.data." + jsonName + "_und_" + region;
+                            orgData = ((outFile[lang][key]) != undefined) ? outFile[lang][key] : {};
+                            parseData = JSON.parse(readData);
+                            mergeData = JSUtils.merge(orgData, parseData, true);
+                            outFile[lang][key] = mergeData;
+                        }
                     }
                 } else if (region) {
-                    jsonPath = path.join(iliblocalePath, lang, region, jsonName + ".json");
+                    jsonPath = path.join(dataPath, lang, region, jsonName + ".json");
                     readData = readFile(jsonPath);
-                    if (readData) outFile[lang]["ilib.data." + jsonName + "_" + lang + "_" + region] = readData;
+                    if (readData) {
+                        key = "ilib.data." + jsonName + "_" + lang + "_" + region;
+                        orgData = ((outFile[lang][key]) != undefined) ? outFile[lang][key] : {};
+                        parseData = JSON.parse(readData);
+                        mergeData = JSUtils.merge(orgData, parseData, true);
+                        outFile[lang][key] = mergeData;
+                    }
 
-                    jsonPath = path.join(iliblocalePath, "und", region, jsonName + ".json");
+                    jsonPath = path.join(dataPath, "und", region, jsonName + ".json");
                     readData = readFile(jsonPath);
-                    if (readData) outFile[lang]["ilib.data." + jsonName + "_und_" + region] = readData;
+                    if (readData) {
+                        key = "ilib.data." + jsonName + "_und_" + region;
+                        orgData = ((outFile[lang][key]) != undefined) ? outFile[lang][key] : {};
+                        parseData = JSON.parse(readData);
+                        mergeData = JSUtils.merge(orgData, parseData, true);
+                        outFile[lang][key] = mergeData;
+                    }
                 }
             } else {
                 console.log("The locale " + lo.getSpec() +  " is missing language code.");
             }
         });
     });
 
-    for (let loc in outFile) {
+    return outFile;
+}
+
+
+function assembleLocale() {
+    let result = {};
+    result = assembleData(path.join(ilibPath, "js/data/locale"), result);
+
+    if (customPath) {
+        result = assembleData(customPath, result);
+    }
+
+    for (let loc in result) {
         let contents = "";
-        for(let keys in outFile[loc]){
-            contents += keys + " = " + outFile[loc][keys] + "\n";
+        for(let keys in result[loc]){
+            contents += keys + " = " + JSON.stringify(result[loc][keys]) + "\n";
```

Comment:
Should this have a semicolon at the end?
```suggestion
            contents += keys + " = " + JSON.stringify(result[loc][keys]) + ";\n";
```

Reply:
I updated it.